### PR TITLE
[WFLY-13625] Security realms should support specifying the charset and encoding for credentials.

### PR DESCRIPTION
**Pull request by SoniaZaldana**
Jira: https://issues.redhat.com/browse/WFLY-13625

Requires: https://issues.redhat.com/browse/WFCORE-5027
WildFly Core PR with tests: https://github.com/wildfly/wildfly-core/pull/4324

### CODE REVIEW

**Review by fjuma — APPROVED** [review]
Just have a small documentation comment but the rest looks good to me. Am going to mark this one as approved.

File: docs/src/main/asciidoc/_elytron/components/JDBC_Security_Realm.adoc [hunk 1]
```diff
@@ -112,6 +112,27 @@ The `scram-mapper` supports the loading of SCRAM passwords which use both a salt
  * `salt-encoding` - The encoding of the salt, either `base64` or `hex`.
  * `iteration-count-index` - The index of the column containing the iteration count.
 
+
+== Hash Character Sets
+
+The various password mappers allow loading multiples values from the database to reconstruct the password.
+This information is then used to hash the client provided password to compare against the password reconstructed from the
+database.
```
[the remark this answers is not in the document]
Comment (fjuma) [reply docs/src/main/asciidoc/_elytron/components/JDBC_Security_Realm.adoc]: Thanks @SoniaZaldana.

**Review by fjuma — APPROVED** [review]

**Comment by darranl** [comment]
@SoniaZaldana as discussed we have run into some failures with the new FilesystemRealmTestCase, we should be ready for you to rebase on master now to reproduce locally.

**Comment by SoniaZaldana** [comment]
Thanks @darranl, I fixed the test failure and rebased.

**Comment by darranl** [comment]
Thank you @SoniaZaldana CI was clean.

**Review by darranl — APPROVED** [review]

**Pull request merged**